clode/cpp/CLODEpython.cpp

```diff
@@ -170,43 +171,84 @@ PYBIND11_MODULE(clode_cpp_wrapper, m) {
     // core clODE solver 
     /****************************************************/
 
-    py::class_<ProblemInfo>(m, "problem_info")
+    py::class_<ProblemInfo>(m, "ProblemInfo")
             .def(py::init<const std::string &,
-                 int,
-                 int,
-                 int,
-                 int,
                  const std::vector<std::string> &,
                  const std::vector<std::string> &,
-                 const std::vector<std::string> &>
-                 ()
-            );
-
-    py::class_<SolverParams<double>>(m, "solver_params")
+                 const std::vector<std::string> &,
+                 int>(),
+                 py::arg("src_file"),
+                 py::arg("vars"),
+                 py::arg("pars"),
+                 py::arg("aux") = std::vector<std::string>(),
+                 py::arg("num_noise") = 1)
```

Comment:
num_noise default can be zero. The C++ layer deals with the case of zero aux or wiener variables already.  See ```CLODE::setNewProblem(ProblemInfo newProb)```

Wiener variables in XPP are usually named, too. We could use exactly the convention you implemented for aux. I'd keep the name as "Wiener" or "wvars" as this refers to a very specific kind of noise simulation.